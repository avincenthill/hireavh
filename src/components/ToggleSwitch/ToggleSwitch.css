.toggle-switch {
  width: var(--s-s);
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  text-align: left;
  height: var(--s-s);
  font-size: var(--fs-xs);
}
.toggle-switch-checkbox {
  display: none;
}
.toggle-switch-label {
  display: block;
  overflow: hidden;
  cursor: pointer;
}
.toggle-switch-inner {
  display: block;
  text-align: center;
  width: 200%;
  margin-left: -100%;
  transition: margin 0.3s ease-in 0s;
}
.toggle-switch-inner::before,
.toggle-switch-inner::after {
  display: block;
  float: left;
  width: 50%;
  height: var(--s-s);
  line-height: 1.95rem;
  box-sizing: border-box;
}
.toggle-switch-inner::after {
  background-color: var(--c-1);
}
.toggle-switch-checkbox:checked + .toggle-switch-label .toggle-switch-inner {
  margin-left: 0;
}
.toggle-switch-checkbox:checked + .toggle-switch-label .toggle-switch-switch {
  right: 0;
}
.toggle-switch-inner::before {
  content: attr(data-yes);
}
.toggle-switch-inner::after {
  content: attr(data-no);
}
