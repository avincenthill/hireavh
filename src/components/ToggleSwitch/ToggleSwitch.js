import React from "react";
import "./ToggleSwitch.css";

class ToggleSwitch extends React.Component {
  state = {
    checked: this.props.defaultChecked,
  };
  onChange = (e) => {
    this.setState({
      checked: e.target.checked,
    });
    if (typeof this.props.onChange === "function") this.props.onChange();
  };
  render() {
    return (
      <div className="toggle-switch">
        <input
          type="checkbox"
          className="toggle-switch-checkbox"
          id={this.props.id}
          checked={this.props.currentValue}
          defaultChecked={this.props.defaultChecked}
          onChange={this.onChange}
        />
        <label className="toggle-switch-label" htmlFor={this.props.id}>
          <span
            className="toggle-switch-inner"
            data-yes={this.props.text[0]}
            data-no={this.props.text[1]}
          ></span>
        </label>
      </div>
    );
  }
}

export default ToggleSwitch;
