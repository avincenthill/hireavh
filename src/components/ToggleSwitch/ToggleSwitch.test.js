import React from "react";
// import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import ToggleSwitch from "./ToggleSwitch";

describe("ToggleSwitch", () => {
  describe("snapshot", () => {
    it("should match the last saved snapshot", () => {
      const tree = renderer.create(<ToggleSwitch text={["0", "1"]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
