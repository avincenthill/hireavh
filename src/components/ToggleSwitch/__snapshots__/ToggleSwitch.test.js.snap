// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch snapshot matches the last saved snapshot 1`] = `
<div
  className="toggle-switch"
>
  <input
    className="toggle-switch-checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="toggle-switch-label"
  >
    <span
      className="toggle-switch-inner"
      data-no="1"
      data-yes="0"
    />
  </label>
</div>
`;
